DependencyInjection/Factory/ChildDefinitionTrait.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\DependencyInjection\Factory;
+
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+
+trait ChildDefinitionTrait
+{
+    /**
+     * @return ChildDefinition|DefinitionDecorator
+     */
+    private function getChildDefinition($parent)
+    {
+        return class_exists('\Symfony\Component\DependencyInjection\ChildDefinition') ?
```

Comment:
Updated all FQCN strings to the special `::class` compile-time constant: https://github.com/liip/LiipImagineBundle/pull/908/commits/ee9c83473de7c4ab57e25fc6b661eb5eb86a0171. Thanks for the recommendation.